src/InternPool.zig

```diff
@@ -54,6 +54,9 @@ string_table: std.HashMapUnmanaged(
     std.hash_map.default_max_load_percentage,
 ) = .{},
 
+/// Source locations of errors being added to an inferred error set.
+ies_src_locs: std.ArrayListUnmanaged(Module.LazySrcLoc) = .{},
```

Comment:
This state being in InternPool is problematic for a bunch of reasons:
* increasing complexity of upcoming incremental compilation changes
* increasing the size of serialized compiler state
* it's long-lived data in memory that is very likely unused

If you can solve this with temporary state in Sema then I think it can be workable, but I think a solution like this should not even be considered until the much higher priority incremental compilation sub-project is complete.

Reply:
I agree it makes sense to prioritize incremental compilation before looking at this change. When we revisit this down the line, it might make sense to instead associate each error in an IES with a ZIR instruction which has a source location attached; this allows us to reuse the existing tracking logic, so solves the incremental update problem, and it's more compact. (But I agree this should be left until incremental is working.)

Reply:
This cannot be handled with temporary state in Sema because the source location (or ZIR instruction) needs to be stored as long as the IES for this to work.